### PR TITLE
fix!: Share splitting bug

### LINKED ISSUE #1108: Bug: v0.10.0 Share splitting with use the incorrect index when many small blobs are submitted
Currently, with each blob submitted, we overestimate the bytes takes up by the actual transaction itself. In some cases, if we have enough blob transactions make it in the same block, then the amount of overestimated shares will be greater than the amount of padding added between the end of the compact shares and start of the sparse shares. This results in using an incorrect share index, and thus proposing an invalid block, validators will validate the share commitments in each PFB using the incorrect starting share index for the first blob. 

This can be fixed by forcing the share splitting logic to use the first share index, instead of calculating it itself.

### PR DESCRIPTION
**Pull request by evan-forbes**
## Overview

fixes the bug described in #1108

the modified tests fails on main. Also see #1110, which we should really get to as soon as we don't have more important things as this code is ancient

closes #1108

## Checklist

- [x] New and updated code has appropriate documentation
- [x] New and updated code has new and/or updated testing
- [x] Required CI checks are passing
- [x] Visual proof for any user facing features like CLI or documentation updates
- [x] Linked issues closed with keywords

### CODE REVIEW

File: pkg/shares/share_splitting.go [hunk 1]
```diff
@@ -45,6 +45,11 @@ func Split(data coretypes.Data, useShareIndexes bool) ([]Share, error) {
 			BlobSharesUsed(len(data.Blobs[0].Data)),
 			int(data.SquareSize),
 		)
+		// force msgIndexes to be the first share index
+		if len(blobIndexes) != 0 && useShareIndexes {
+			blobShareStart = int(blobIndexes[0])
+		}
```
Comment (cmwaters) [inline pkg/shares/share_splitting.go:51]: Are we able to specifically add a unit test for this?

File: pkg/shares/share_splitting.go [hunk 2]
```diff
@@ -45,6 +45,11 @@ func Split(data coretypes.Data, useShareIndexes bool) ([]Share, error) {
 			BlobSharesUsed(len(data.Blobs[0].Data)),
 			int(data.SquareSize),
 		)
+		// force msgIndexes to be the first share index
```
Comment (rootulp) [inline pkg/shares/share_splitting.go:48]:
```suggestion
		// force blobSharesStart to be the first share index
```

**Review by rootulp — DISMISSED** [review]

**Review by rahulghangas — DISMISSED** [review]

File: pkg/shares/share_splitting.go (hunk 1 above)
Comment (evan-forbes) [reply pkg/shares/share_splitting.go]: :+1: yeah defo needed, thanks. fails on main
https://github.com/celestiaorg/celestia-app/pull/1111/commits/c67976551ac4cebc636bef35ea73ff1ffca65bb2

also https://github.com/celestiaorg/celestia-app/issues/1110#issuecomment-1352389547

File: pkg/shares/shares_test.go [hunk 3]
```diff
@@ -62,6 +64,32 @@ func TestMerge(t *testing.T) {
 	}
 }
 
+// TestPadFirstIndexedBlob ensures that we are adding padding to the first share
+// instead of calculating the value.
```
Comment (rootulp) [inline pkg/shares/shares_test.go:68]: [non-blocking] sorry I don't understand what "calculating the value" means in this context. Do you mind elaborating? It doesn't strictly have to go into the comment b/c I don't want to block this PR's merge.

**Review by rootulp — APPROVED** [review]

File: pkg/shares/shares_test.go (hunk 3 above)
Comment (evan-forbes) [reply pkg/shares/shares_test.go:68]: calculate using the non-interactive defaults, previously this is what it was doing. The estimation process was accurate enough that this was the case the vast majority of the time, but not for all cases.

**Pull request merged**